cardano-config/src/Cardano/Config/ShelleyGenesis.hs

```diff
@@ -0,0 +1,235 @@
+{-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings          #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Cardano.Config.ShelleyGenesis
+  ( ShelleyGenesis(..)
+  , shelleyGenesisDefaults
+  , shelleyGenesisToJSON
+  , shelleyGenesisFromJSON
+  ) where
+
+import           Cardano.Prelude
+
+import qualified Data.Text.Encoding as Text
+import qualified Data.Map.Strict as Map
+import qualified Data.Time as Time
+
+import           Data.Aeson (Value, ToJSON(..), toJSON, (.=),
+                             FromJSON(..), (.:),
+                             ToJSONKey(..),   ToJSONKeyFunction(..),
+                             FromJSONKey(..), FromJSONKeyFunction(..))
+import           Data.Aeson.Types    (Parser)
+import qualified Data.Aeson          as Aeson
+import qualified Data.Aeson.Encoding as Aeson
+
+import           Control.Monad.Fail
+
+import           Cardano.Crypto.Hash.Class as Crypto
+import           Cardano.Crypto.ProtocolMagic (ProtocolMagicId(..))
+import           Cardano.Slotting.Slot (EpochSize (..))
+
+import           Ouroboros.Network.Magic (NetworkMagic (..))
+import           Ouroboros.Consensus.BlockchainTime
+                   (SlotLength (..), SystemStart (..), slotLengthFromSec)
+import           Ouroboros.Consensus.Protocol.Abstract (SecurityParam (..))
+import           Ouroboros.Consensus.Shelley.Node (ShelleyGenesis (..))
+import           Ouroboros.Consensus.Shelley.Protocol (TPraosStandardCrypto)
+
+
+import           Shelley.Spec.Ledger.Address (serialiseAddr, deserialiseAddr)
+import           Shelley.Spec.Ledger.Coin   (Coin(..))
+import           Shelley.Spec.Ledger.Crypto (Crypto)
+import           Shelley.Spec.Ledger.Keys   (HashAlgorithm)
+import           Shelley.Spec.Ledger.Keys   (DiscKeyHash(..))
+import           Shelley.Spec.Ledger.TxData (Addr(..))
+
+
+
+-- | Some reasonable starting defaults for constructing a 'ShelleyGenesis'.
+--
+-- You must override at least the following fields for this to be useful:
+--
+-- * 'sgStartTime' the time of the first block
+-- * 'sgNetworkMagic' to a suitable testnet or mainnet network magic number.
+-- * 'sgGenDelegs' to have some initial nodes
+-- * 'sgInitialFunds' to have any money in the system
+-- * 'sgMaxLovelaceSupply' must be at least the sum of the 'sgInitialFunds'
+--   but more if you want to allow for rewards.
+--
+shelleyGenesisDefaults :: ShelleyGenesis crypto
+shelleyGenesisDefaults =
+  ShelleyGenesis
+    { sgStartTime         = SystemStart zeroTime
+    , sgNetworkMagic      = NetworkMagic 0
+    , sgProtocolMagicId   = ProtocolMagicId 0
+
+    , sgSlotLength        = slotLengthFromSec 1 -- 1s slots
+    , sgActiveSlotsCoeff  = 1/20                -- 20s block times on average
+    , sgDecentralisationParam = 1
```

Comment:
Can we either align all of them or none? :grin:

Reply:
My vote would be to align none.

Reply:
I agree with no alignment, the majority of the repo is already like this.